src/Nethermind/Nethermind.Optimism/CL/Decoding/DecodingPipeline.cs

```diff
@@ -69,4 +78,28 @@ public async Task Run(CancellationToken token)
             if (logger.IsInfo) logger.Info($"Decoding pipeline is shutting down.");
         }
     }
+
+    private async Task Clear(CancellationToken token)
+    {
+        while (await _inputChannel.Reader.WaitToReadAsync(token))
+        {
+            while (_inputChannel.Reader.TryRead(out _)) { }
+        }
+
+        while (await _outputChannel.Reader.WaitToReadAsync(token))
+        {
+            while (_outputChannel.Reader.TryRead(out _)) { }
+        }
```

Comment:
We're draining the channels here but I don't see how we're preventing other components from continue writing (is this OK behavior)? For example:

https://github.com/NethermindEth/nethermind/blob/4b898515feef499e63a61bf918cdc7d7eecbe76c/src/Nethermind/Nethermind.Optimism/CL/Driver.cs#L84